src/unix/solarish/mod.rs

```diff
@@ -460,6 +462,38 @@ s! {
         pub pi_fputypes: [::c_char; PI_FPUTYPE as usize],
         pub pi_clock: ::c_int,
     }
+
+    pub struct prmap_t {
+        pub pr_vaddr: ::uintptr_t,
+        pub pr_size: ::size_t,
+        pub pr_mapname: [[::c_char; 32]; 2],
```

Comment:
Why did this change from `[::c_char; PRMAPSZ as usize]`?

Reply:
the failure might have to do with the fact it s an array sized > 32 (happened before).

Reply:
If definitions implied by the `s!` macro cannot handle the >32 length, we should move this struct outside that macro, rather than altering the type.

Reply:
@jclulow see for ex https://github.com/rust-lang/libc/blob/8b05ed7ebb9cf24a3473764a3dca3f9f3397477f/src/unix/bsd/netbsdlike/netbsd/mod.rs#L514 there are couple more examples here and there.

Reply:
I agree with @pfmooney here, I think it would be better to disable types that can't be expressed precisely on older rustc versions.

Reply:
> I agree with @pfmooney here, I think it would be better to disable types that can't be expressed precisely on older rustc versions.

We could also just provide our own impls of `Eq` and `Hash` which handled the long array members here.

Reply:
That s what I started to do in fact.